### ISSUE TITLE
Detect incompatible fields in `TypedDict` multiple inheritance

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#6589]
If a `TypedDict` class derives from two or more other `TypedDict` classes and there is a field defined in multiple base classes, pyright should enforce that the `Required` / `NotRequired` attribute matches.

```python
from typing import NotRequired, Required, TypedDict

class A(TypedDict):
    x: Required[int]

class B(TypedDict):
    x: NotRequired[int]

class C(A, B):  # Should generate an error
    ...
```

**Comment by alicederyn** [comment]
Related case to consider:

```python
from typing import NotRequired, Required, TypedDict

class A(TypedDict):
    x: ReadOnly[Required[int]]

class B(TypedDict):
    x: ReadOnly[NotRequired[int]]

class C(A, B):
    pass
```

I think in this case, there's no error and the key ends up read-only and required.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.339, which I just published. It will also be included in a future release of pylance.
